app/uk/gov/hmrc/crdlcachestub/repositories/migration/ImportCodeListsMigration.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2025 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.crdlcachestub.repositories.migration
+
+import javax.inject.Singleton
+import uk.gov.hmrc.crdlcachestub.repositories.CodeListsRepository
+import uk.gov.hmrc.crdlcachestub.models.CodeListEntry
+import org.apache.pekko.stream.scaladsl.StreamConverters
+import org.apache.pekko.stream.scaladsl.JsonFraming
+
+import javax.inject.Inject
+import org.apache.pekko.stream.Materializer
+import uk.gov.hmrc.mongo.transaction.Transactions
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.transaction.TransactionConfiguration
+import org.apache.pekko.actor.ActorSystem
+
+import scala.concurrent.ExecutionContext
+import play.api.libs.json.Json
+import play.api.libs.json.JsResult
+
+import scala.concurrent.Future
+import play.api.Logging
+import uk.gov.hmrc.crdlcachestub.models.formats.MongoFormats
+
+@Singleton
+class ImportCodeListsMigration @Inject() (
+  val mongoComponent: MongoComponent,
+  repository: CodeListsRepository,
+  resourceProvider: ResourceProvider
+)(using system: ActorSystem)
+  extends Transactions
+  with Logging
+  with MongoFormats {
+
+  given ExecutionContext         = system.dispatcher
+  given TransactionConfiguration = TransactionConfiguration.strict
+
+  val migrationComplete: Future[Unit] =
```

Comment:
This migration pattern might have a potential issue where failed migrations can cause requests to hang indefinitely. If I understood correctly, if the migration Future fails, the error is only logged, but the Future never completes and controllers waiting on migrationComplete will hang indefinitely.

Could do something like:
```scala
val migrationComplete: Future[Unit] = 
  withSessionAndTransaction { ... }.recoverWith {
    case ex => 
      logger.error("Migration failed...", ex)
      System.exit(1)
      Future.failed(ex)
  }
```

Lazy loading could work too?
```scala
lazy val migrationComplete: Future[Unit] = withSessionAndTransaction { ... }
```